### ISSUE TITLE
Agent replies to external users are creating new conversations instead of threading into the existing

### ISSUE THREAD

**Issue by ke1v3y** [osTicket/osTicket#5249]
**Summary**
When replying to a ticket, it is expected that the conversation with the customer (external to the system) stays threaded in their email inbox, in addition to remaining threaded in OS Ticket. However, while the conversation remains in tact internally, externally, both Gmail and Outlook treat subsequent replies by an agent to an external party to be a new conversation.

Note that this has been previously reported as https://github.com/osTicket/osTicket/issues/1792 but is still without resolution. Additionally, I am not seeing any activity on the outstanding issue.

**Server/Application Versions**
osTicket				1.12.3
Apache Version 		2.4.41
PHP Version			7.2.25
MySQL Version		10.1.41-MariaDB
Architecture			x86_64
Operating System		linux
Perl Version			5.16.3
Kernel Version			3.10.0-957.12.2.vz7.96.21

![MicrosoftTeams-image](https://user-images.githubusercontent.com/42303322/70187332-eaab1f00-16bb-11ea-9c96-4622cf9506ee.png)

**Comment by JediKev** [comment]
@ke1v3y 

We are seeing new reports now as well about this issue. So we are taking steps to look into and address this.

Cheers.

**Comment by JediKev** [comment]
@ke1v3y 

Note there has been activity to try and address this with `MailingList` but it appears this is yet another specific instance:
- https://github.com/osTicket/osTicket/pull/5138

Cheers.

**Comment by ke1v3y** [comment]
@JediKev -

Thank you for the swift reply and additional information. It looks like this fix is in the latest release (1.14.1) on osTicket's website. Is this considered a stable release?

Best regards.

**Comment by JediKev** [comment]
@ke1v3y 

Yes, v1.14.1 is considered stable. Anything that is not a release candidate on our website is considered stable. As far as the commit being in v1.14.1, yes it is included. We have received new reports of non-threading responses though after the commit being applied so it might be another specific case that we are currently looking into. Any updates will be posted here.

Cheers.

**Comment by ke1v3y** [comment]
@JediKev 

We have updated to the latest build of osTicket (v1.14.1) and are now seeing threading on replies! However, we still are not seeing threading on the system-generated auto-reply to the initial request. Please let us know if we can provide any information to facilitate debugging.

Thank you.

**Comment by JediKev** [comment]
@ke1v3y 

In my testing I see this as well _but_ the email does have the appropriate In-Reply-To and References headers and they do match the original MessageID so I'm thinking at this point it's a mail provider/client thing.

Cheers.
